lib/rouge/lexers/jsl.rb

```diff
@@ -14,24 +14,24 @@ class JSL < RegexLexer
         rule %r/\s+/m, Text::Whitespace
 
         rule %r(//.*?$), Comment::Single
-        rule %r'/[*].*', Comment::Multiline, :comment
+        rule %r'/[*].*[*]/', Comment::Multiline # single line block comment
+        rule %r'/[*].*', Comment::Multiline, :comment # multiline block comment
 
         # messages
         rule %r/<</, Operator, :message
 
         # covers built-in and custom functions
-        rule %r/([a-z_][\w\s'%.\\]*)(\()/i do |m|
-          groups Keyword, Punctuation
+        rule %r/(::|:)?([a-z_][\w\s'%.\\]*)(\()/i do |m|
+          groups Punctuation, Keyword, Punctuation
         end
 
         rule %r/\d{2}(jan|feb|mar|apr|may|jun|jul|aug|sep|oct|nov|dec)\d{2}(\d{2})?(:\d{2}:\d{2}(:\d{2}(\.\d*)?)?)?/i, Literal::Date
 
         rule %r/-?(?:[0-9]+(?:[.][0-9]+)?|[.][0-9]*)(?:e[+-]?[0-9]+)?i?/i, Num
 
-        rule %r/::[a-z_][\w\s'%.\\]*/i, Name::Variable
-        rule %r/:\w+/, Name
-        rule %r/[a-z_][\w\s'%.\\]*/i, Name::Variable
-        rule %r/"(?:\\!"|[^"])*?"n/m, Name::Variable
+        rule %r/(::|:)?([a-z_][\w\s'%.\\]*|"(?:\\!"|[^"])*?"n)/i do |m|
+          groups Punctuation, Name::Variable
+        end
```

Comment:
Is it desirable for the `::` or `:` to be styled/treated differently than the name variable itself? For example, can we simply this rule to the following?

```ruby
rule %r/(::|:)?([a-z_][\w\s'%.\\]*|"(?:\\!"|[^"])*?"n)/i, Name::Variable
```

Reply:
I would consider the colons separate from the name. I think it helps to visually separate the namespace from the variable name. For example in the visual sample I just added `rouge:scoped.func = ...` where `rouge` is the namespace.   
![image](https://user-images.githubusercontent.com/6343165/105193450-8426d180-5b06-11eb-8318-867a655d443b.png)

Reply:
I reached out to some colleagues to see if anyone are familiar with JSL to give some feedback. From my Ruby experience this doesn't make sense. In Ruby `::Foo` or `Foo::Bar` the colons are considered part of the object and have always been highlighted together in every case I've seen even outside Rouge. Maybe this is a language-specific thing, though.

Reply:
I don't have a super strong opinion, so if we don't get further feedback shortly, I'll accept as-is. We can always adjust later if anyone disagrees.